### PR TITLE
[CodeCompletion] Remove static member completions from metatypes

### PR DESCRIPTION
**Pull request by AnthonyLatsis**
Resolves [SR-5458](https://bugs.swift.org/browse/SR-5458).

### CODE REVIEW

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test

**Comment by AnthonyLatsis** [comment]
Sorry, I was a few seconds late with the update

**Comment by jrose-apple** [comment]
@swift-ci Please smoke test

**Review by benlangmuir — CHANGES_REQUESTED** [review]
Do we get any completions on `Foo.Type` after this?

**Comment by AnthonyLatsis** [comment]
> Do we get any completions on Foo.Type after this?

on any expression which has a type of higher order than `Foo.Type`, i.e. `Foo.Type.Type`, we only show `self` and **should** show`Type`, otherwise we do as always.

File: test/IDE/complete_value_expr.swift [hunk 1]
```diff
@@ -497,6 +499,26 @@ func testMetatypeExprWithoutDot() {
...
+  NoMetaCompletions.Type#^FOO_STRUCT_META_2^#
+}
+// FOO_STRUCT_META-NOT: Decl[StaticVar]/CurrNominal:    .foo[#Int#]; name=foo
+// FOO_STRUCT_META-NOT: Decl[StaticMethod]/CurrNominal: .bar()[#Void#]; name=bar()
+// FOO_STRUCT_META-NOT: Decl[TypeAlias]/CurrNominal:    .Foo[#Int#]; name=Foo
+// FOO_STRUCT_META-NOT: Decl[Constructor]/CurrNominal:  ()[#NoMetaCompletions#]; name=()
+// FOO_STRUCT_META-NOT: Decl[StaticVar]/CurrNominal:    foo[#Int#]; name=foo
+// FOO_STRUCT_META-NOT: Decl[StaticMethod]/CurrNominal: bar()[#Void#]; name=bar()
+// FOO_STRUCT_META-NOT: Decl[TypeAlias]/CurrNominal:    Foo[#Int#]; name=Foo
+// FOO_STRUCT_META-NOT: Decl[Constructor]/CurrNominal:  init()[#NoMetaCompletions#]; name=init()
```
[the remark this answers is not in the document]
Comment (AnthonyLatsis) [reply test/IDE/complete_value_expr.swift]: Oh, great! I didn't know FileCheck does partial matches by default.

**Comment by benlangmuir** [comment]
> we only show `self` and should show `Type`

Let's add to your test that `self` shows up then.  You could do `FOO_STRUCT_META-NOT: Decl` before and after the check for self.

**Comment by benlangmuir** [comment]
@swift-ci please smoke test

**Review by benlangmuir — APPROVED** [review]

**Pull request merged**